src/app/clusters/mode-select-server/mode-select-server.cpp

```diff
@@ -109,35 +114,201 @@ CHIP_ERROR ModeSelectAttrAccess::Read(const ConcreteReadAttributePath & aPath, A
     return CHIP_NO_ERROR;
 }
 
-} // anonymous namespace
-
-bool emberAfModeSelectClusterChangeToModeCallback(CommandHandler * commandHandler, const ConcreteCommandPath & commandPath,
-                                                  const ModeSelect::Commands::ChangeToMode::DecodableType & commandData)
+Status ChangeToMode(EndpointId endpointId, uint8_t newMode)
 {
     MATTER_TRACE_SCOPE("ChangeToMode", "ModeSelect");
     ChipLogProgress(Zcl, "ModeSelect: Entering emberAfModeSelectClusterChangeToModeCallback");
-    EndpointId endpointId = commandPath.mEndpointId;
-    uint8_t newMode       = commandData.newMode;
+
     // Check that the newMode matches one of the supported options
     const ModeSelect::Structs::ModeOptionStruct::Type * modeOptionPtr;
     const ModeSelect::SupportedModesManager * gSupportedModeManager = ModeSelect::getSupportedModesManager();
     if (gSupportedModeManager == nullptr)
     {
         ChipLogError(Zcl, "ModeSelect: SupportedModesManager is NULL");
-        commandHandler->AddStatus(commandPath, Status::Failure);
-        return true;
+        return Status::Failure;
     }
     Status checkSupportedModeStatus = gSupportedModeManager->getModeOptionByMode(endpointId, newMode, &modeOptionPtr);
     if (Status::Success != checkSupportedModeStatus)
     {
         ChipLogProgress(Zcl, "ModeSelect: Failed to find the option with mode %u", newMode);
-        commandHandler->AddStatus(commandPath, checkSupportedModeStatus);
-        return true;
+        return checkSupportedModeStatus;
     }
     ModeSelect::Attributes::CurrentMode::Set(endpointId, newMode);
 
+    return Status::Success;
+}
+
+} // anonymous namespace
+
+#if defined(MATTER_DM_PLUGIN_SCENES_MANAGEMENT) && CHIP_CONFIG_SCENES_USE_DEFAULT_HANDLERS
```

Comment:
could we find a way to limit ifdef code management? Can this be part of a build system config that compiles or does not compile specific parts of the code?

Reply:
`CHIP_CONFIG_SCENES_USE_DEFAULT_HANDLERS ` Could be part of a build config, but it is currently being used for all the other cluster's handlers as well so I'd rather move this to another PR.

Essentially the scenes cluster needs this define to work out of the box, so I could build a : `use_custom_scenes_handler` build flag to set this config to false.

Reply:
Opened issue here: https://github.com/project-chip/connectedhomeip/issues/36102